shared/desktop/remote/sync-browser-window.desktop.tsx

```diff
@@ -13,119 +10,35 @@ type Props = {
   windowParam: string
 }
 
-const defaultWindowOpts = {
-  frame: false,
-  fullscreen: false,
-  height: 300,
-  resizable: false,
-  show: false, // Start hidden and show when we actually get props
-  titleBarStyle: 'customButtonsOnHover' as const,
-  webPreferences: {
-    nodeIntegration: true,
-    nodeIntegrationInWorker: false,
-  },
-  width: 500,
-}
-
-type State = {
-  remoteWindow: SafeElectron.BrowserWindowType | null
-}
-
 function SyncBrowserWindow(ComposedComponent: any) {
-  class RemoteWindowComponent extends React.PureComponent<Props, State> {
-    _remoteWindow: SafeElectron.BrowserWindowType | null = null
-    _remoteWindowId: number | null = null
-    _mounted: boolean = false
-
-    // We only have state to force re-renders and to pass down to the child. We usually want to just use the raw _remoteWindow to avoid races
-    state = {
-      remoteWindow: null,
-    }
-
-    _makeBrowserWindow = (): SafeElectron.BrowserWindowType => {
-      const windowOpts = {
-        ...defaultWindowOpts,
-        ...this.props.windowOpts,
-      }
-      const w = new SafeElectron.BrowserWindow(windowOpts)
-      this._remoteWindow = w
-      if (this._mounted) {
-        this.setState({remoteWindow: this._remoteWindow})
-      }
-      this._positionBrowserWindow(windowOpts)
-      return w
-    }
-
-    _positionBrowserWindow = (windowOpts: {width: number; height: number}) => {
-      if (this.props.windowPositionBottomRight && SafeElectron.getScreen().getPrimaryDisplay()) {
-        const {width, height} = SafeElectron.getScreen().getPrimaryDisplay().workAreaSize
-        this._remoteWindow &&
-          this._remoteWindow.setPosition(
-            width - windowOpts.width - 100,
-            height - windowOpts.height - 100,
-            false
-          )
-      }
-    }
-
-    _setupWebContents = () => {
-      if (!this._remoteWindow) {
-        return
-      }
-      const webContents = this._remoteWindow.webContents
-      if (showDevTools && !skipSecondaryDevtools) {
-        webContents.openDevTools({mode: 'detach'})
+  const RemoteWindowComponent = (props: Props) => {
+    React.useEffect(() => {
+      const {windowOpts, windowComponent, windowParam, windowPositionBottomRight} = props
+
+      SafeElectron.getApp().emit('KBkeybase', '', {
```

Comment:
we ask the node thread to make and destroy this window when this component is mounted/unmounted (vs doing it ourselves before)